x-pack/legacy/plugins/siem/public/components/flyout/button/index.tsx

```diff
@@ -144,7 +145,7 @@ export const FlyoutButton = React.memo<FlyoutButtonProps>(
     ) : null,
   (prevProps, nextProps) =>
     prevProps.show === nextProps.show &&
-    prevProps.dataProviders === nextProps.dataProviders &&
```

Comment:
if there is not a bug here, please revert this to a reference comparison (here, and anywhere else which a deep comparison is unnecessary)

Reply:
![image](https://user-images.githubusercontent.com/5188868/76451614-f94a5600-63cf-11ea-8b9d-c86dce85ee5b.png)
I think it's necessary here to avoid unnecessary re-renders

Reply:
Thanks for the clarification! For anyone following this thread: we tracked down (over a zoom) the root cause of the unnecessary re-render:  a `lodash/fp` `getOr` was returning a new reference in a `mapStateToProps`. Our plan here is to eliminate the `getOr` (replacing it with the `??` operator, which also gives us better type saftey), and drop the proposed `deepEqual` in favor of the reference comparison.